### PR TITLE
Add user friendly view of storage data 

### PR DESCRIPTION
**Pull request by smiasojed**
## Summary
Related #101
- [ ] y/n | Does it introduce breaking changes?
- [ ] y/n | Is it dependent on the specific version of `ink` or `pallet-contracts`?

Add a user-friendly view of contract storage data in the form of a table

## Description

Example output:

Screenshot 2023-12-11 at 10 54 09

Screenshot 2023-12-11 at 11 09 09

## Checklist before requesting a review
- [x] My code follows the style guidelines of this project
- [ ] I have added an entry to `CHANGELOG.md`
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ascjones — COMMENTED** [review]
WIP review, will finish when I return from Christmas!

File: crates/transcode/src/decode.rs [hunk 1]
```diff
@@ -177,7 +177,7 @@ impl<'a> Decoder<'a> {
         variant_type: &TypeDefVariant<PortableForm>,
         input: &mut &[u8],
     ) -> Result<Value> {
-        let discriminant = input.read_byte()?;
+        let discriminant = input.read_byte().unwrap_or(0);
```
Comment (ascjones) [inline crates/transcode/src/decode.rs:180]: If it is an enum then the discriminant should always be the first byte.

**Review by ascjones — COMMENTED** [review]
Overall looks good. `contract_storage.rs` is starting to be quite a large file, so might benefit from breaking up files for different concerns e.g. fetching vs displaying the data.

File: crates/cargo-contract/src/cmd/storage.rs [hunk 2]
```diff
@@ -105,4 +110,29 @@ impl StorageCommand {
...
+            "Parent".bright_purple().bold(),
+            "Value".bright_purple().bold()
+        );
+
+        for (index, cell) in storage.iter().enumerate() {
+            let decoded_cell = cell.decode_pretty_string();
+            let values = decoded_cell.split('\n');
+            for (i, v) in values.enumerate() {
+                println!(
+                    "{:<5} | {:<8} | {:<20.20} | {}",
```
[the remark this answers is not in the document]
Comment (smiasojed) [reply crates/cargo-contract/src/cmd/storage.rs]: Done

File: crates/extrinsics/src/contract_storage.rs [hunk 3]
```diff
@@ -76,10 +94,25 @@ where
...
-        let storage_keys = self
-            .rpc
-            .fetch_storage_keys_paged(trie_id, None, 1000, None, None) // todo loop pages
-            .await?;
+        let mut storage_keys = Vec::new();
+        const KEYS_COUNT: u32 = 1000;
+        loop {
+            let mut keys = self
+                .rpc
+                .fetch_storage_keys_paged(
```
[the remark this answers is not in the document]
Comment (smiasojed) [reply crates/extrinsics/src/contract_storage.rs:103]: Done

File: crates/transcode/src/decode.rs (hunk 1 above)
Comment (smiasojed) [reply crates/transcode/src/decode.rs:180]: Is it possible to have a 0-byte enum, where the discriminant is missing? I faced this issue while testing the storage.

Comment (ascjones) [reply crates/transcode/src/decode.rs:180]: Ok, yes it is possible if the enum has no variants at all: https://github.com/paritytech/parity-scale-codec/blob/master/derive/src/encode.rs#L314. 

So looks like we would just return an error from here in that case? https://github.com/paritytech/cargo-contract/blob/890d4272accd1de96828cb88e3dc04042ea8b396/crates/transcode/src/decode.rs#L186

File: crates/extrinsics/src/contract_storage.rs [hunk 4]
```diff
@@ -101,73 +134,410 @@ where
...
+impl RootKeyEntry {
+    fn key_as_hex<S>(key: &u32, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(format!("0x{}", hex::encode(key.to_le_bytes())).as_str())
+    }
+}
+
+trait DecodePrettyString {
```
[the remark this answers is not in the document]
Comment (smiasojed) [reply crates/extrinsics/src/contract_storage.rs]: Done

File: crates/transcode/src/decode.rs (hunk 1 above)
Comment (smiasojed) [reply crates/transcode/src/decode.rs:180]: I am not sure. If it is allowed, why would we want to return an error while decoding storage?

File: crates/extrinsics/src/contract_storage.rs [hunk 5]
```diff
@@ -101,73 +134,410 @@ where
...
+    /// Decode the storage cell into a multiline string.
+    fn decode_pretty_string(&self) -> String;
+}
+
+#[derive(Serialize, Debug)]
+pub struct Mapping {
+    #[serde(flatten)]
+    root: RootKeyEntry,
+    #[serde(serialize_with = "Mapping::map_as_vec")]
```
[the remark this answers is not in the document]
Comment (smiasojed) [reply crates/extrinsics/src/contract_storage.rs]: Done

File: crates/transcode/src/decode.rs (hunk 1 above)
Comment (smiasojed) [reply crates/transcode/src/decode.rs]: You are correct; this was a side issue of decoding `Cell<Option>`, which is not present in the storage but is defined in the type definition.

**Review by ascjones — APPROVED** [review]
LGTM. Can you add to `CHANGELOG` please?

**Pull request merged**